python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -351,6 +363,20 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
     def get_autoscaling_actor_pools(self) -> List[AutoscalingActorPool]:
         return [self._actor_pool]
 
+    def update_resource_usage(self) -> None:
+        """Updates resources usage."""
+        actors = list(self._actor_pool._num_tasks_in_flight.keys())
+        for actor in actors:
+            actor_state = actor._get_local_state()
+            if actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE:
```

Comment:
let's add an assertion here to make sure we are only handling RESTARTING state here.